be/src/service/internal_service.cpp

```diff
@@ -1498,7 +1498,7 @@ void PInternalServiceImpl::request_slave_tablet_pull_rowset(
                                     rowset_meta->tablet_id(), node_id, true);
     });
     if (!ret) {
-        offer_failed(response, done, _heavy_work_pool);
```

Comment:
not modify like this.

move  brpc::ClosureGuard closure_guard(done); to try offer code.

Reply:
Unlike other RPC impl, true response of pull rowset operation is sent to the caller by a new RPC called `response_slave_tablet_pull_rowset`. So here, we need to acknowledge the current `request_slave_tablet_pull_rowset` ASAP.  We do need more discussion here to decide how we can fix it decently (and more on the thread pool single replica load is using). Let's freeze this PR for a while.